app/src/main/java/org/session/libsession/messaging/open_groups/OpenGroupApi.kt

```diff
@@ -498,108 +490,72 @@ object OpenGroupApi {
             endpoint = Endpoint.RoomMessage(room),
             parameters = parameters
         )
-        return getResponseBodyJson(request, signRequest = true).map { json ->
-            @Suppress("UNCHECKED_CAST") val rawMessage = json as? Map<String, Any>
-                ?: throw Error.ParsingFailed
-            val result = OpenGroupMessage.fromJSON(rawMessage) ?: throw Error.ParsingFailed
-            val storage = MessagingModuleConfiguration.shared.storage
-            storage.addReceivedMessageTimestamp(result.sentTimestamp)
-            result
-        }
+        val json =  getResponseBodyJson(request, signRequest = true)
+        @Suppress("UNCHECKED_CAST") val rawMessage = json as? Map<String, Any>
+            ?: throw Error.ParsingFailed
+        val result = OpenGroupMessage.fromJSON(rawMessage) ?: throw Error.ParsingFailed
+        val storage = MessagingModuleConfiguration.shared.storage
+        storage.addReceivedMessageTimestamp(result.sentTimestamp)
+        return result
     }
     // endregion
 
-    fun addReaction(room: String, server: String, messageId: Long, emoji: String): Promise<AddReactionResponse, Exception> {
+    @OptIn(ExperimentalSerializationApi::class)
+    suspend fun addReaction(room: String, server: String, messageId: Long, emoji: String): AddReactionResponse {
         val request = Request(
             verb = PUT,
             room = room,
             server = server,
             endpoint = Endpoint.Reaction(room, messageId, emoji),
             parameters = emptyMap<String, String>()
         )
-        val pendingReaction = PendingReaction(server, room, messageId, emoji, true)
-        return getResponseBody(request, signRequest = true).map { response ->
-            JsonUtil.fromJson(response, AddReactionResponse::class.java).also {
-                val index = pendingReactions.indexOf(pendingReaction)
-                pendingReactions[index].seqNo = it.seqNo
-            }
-        }
+
+        val response = getResponseBody(request, signRequest = true)
+        val reaction: AddReactionResponse = MessagingModuleConfiguration.shared.json.decodeFromStream(response.inputStream())
```

Comment:
I think we should close the input stream, even it's bytearray's input stream (same for the `deleteReaction`)


```suggestion
        val reaction: AddReactionResponse =  response.inputStream().use( MessagingModuleConfiguration.shared.json::decodeFromStream)
```